astrowidgets/interface_definition.py

```diff
@@ -0,0 +1,111 @@
+from typing import Protocol, runtime_checkable, Any
+from abc import abstractmethod
+
+# Allowed locations for cursor display
+ALLOWED_CURSOR_LOCATIONS = ['top', 'bottom', None]
+
+# List of marker names that are for internal use only
+RESERVED_MARKER_SET_NAMES = ['all']
+
+__all__ = [
+    'ImageViewerInterface',
+    'ALLOWED_CURSOR_LOCATIONS',
+    'RESERVED_MARKER_SET_NAMES'
+]
+
+
+@runtime_checkable
+class ImageViewerInterface(Protocol):
+    # This are attributes, not methods. The type annotations are there
+    # to make sure Protocol knows they are attributes. Python does not
+    # do any checking at all of these types.
```

Comment:
We could enforce it with third-party type checker package but is that going too far?

Reply:
Hmm, although it's a bit odd, when I think about it I like the cleanness of the interface not doing any checking but the APITest class doing it. So maybe if we implemented type-checking it could use this information but actually do the type checking when the APITest is invoked by a backend?

Reply:
Agreed -- and the way we have set things up I don't think we can.